cmd/snap-failure/cmd_snapd.go

```diff
@@ -126,6 +126,11 @@ func (c *cmdSnapd) Execute(args []string) error {
 		// system, either a remodel or a plain snapd installation, call
 		// the snapd from the core snap
 		snapdPath = filepath.Join(dirs.SnapMountDir, "core", "current", "/usr/lib/snapd/snapd")
+		if !osutil.FileExists(snapdPath) {
+			// it is possible that the core snap is not installed at
+			// all, in which case we should try the snapd snap
+			snapdPath = filepath.Join(dirs.SnapMountDir, "snapd", "current", "/usr/lib/snapd/snapd")
```

Comment:
hmm I'm concerned about this fallback of attempting to execute the snapd snap, since in this case, we either had a broken snapd from outside of snap packaging (as in the case of the bug, in which case this fallback is fine), or the snapd snap was the only version of snapd available on this system and there was only one version, and it crashed and so snap-failure tries to execute that one again. But if it was broken and now we are executing the same one again, we run the risk of being stuck in an infinite loop, no?